training/tests/unit/diagnostics/callbacks/test_variable_order.py

```diff
@@ -44,215 +45,142 @@ def name_to_index_rename_permute() -> dict:
 @pytest.fixture
 def fake_trainer(mocker: Any, name_to_index: dict) -> AnemoiTrainer:
     trainer = mocker.Mock(spec=AnemoiTrainer)
```

Comment:
Could it be possible to have a test with `load_weights_only` True, and another one with transfer learning? Asking, cause I am assume the ones here currently are based on the case on resuming/forking runs, but let me know if that's not the case